TokenSale.sol

```diff
@@ -85,12 +91,24 @@ contract TokenSale is TokenSaleInterface, Token {
 
     function refund() noEther {
         if (now > closingTime
-            && !isFunded
-            && msg.sender.call.value(balances[msg.sender])()) // execute refund
+            && !isFunded)
         {
-            Refund(msg.sender, balances[msg.sender]);
-            totalSupply -= balances[msg.sender];
-            balances[msg.sender] = 0;
+            // get extraBalance - will only succeed when called for the first time
+            extraBalance.payOut(address(this), extraBalance.accumulatedInput());
+
+            // execute refund
+            if (msg.sender.call.value(weiGiven[msg.sender])()) {
+                Refund(msg.sender, weiGiven[msg.sender]);
+                totalSupply -= balances[msg.sender];
+                balances[msg.sender] = 0;
+            }
         }
     }
+
+    function divisor() returns (uint divisor){
+        if (closingTime - 2 weeks > now) return 20;
```

Comment:
Definitely add comments here in all 3 legs of the if explaining the different periods and the pricing increase.